pkg/notes/notes.go

```diff
@@ -1019,59 +1020,85 @@ func (g *Gatherer) prsForCommitFromSHA(sha string) (prs []*gogithub.PullRequest,
 }
 
 func (g *Gatherer) prsForCommitFromMessage(commitMessage string) (prs []*gogithub.PullRequest, err error) {
-	prsNum, err := prsNumForCommitFromMessage(commitMessage)
+	mainPrNum, err := prNumForCommitFromMessage(commitMessage)
 	if err != nil {
 		return nil, err
 	}
-	var res *gogithub.PullRequest
-	var resp *gogithub.Response
 
-	for _, pr := range prsNum {
-		// Given the PR number that we've now converted to an integer, get the PR from
-		// the API
-		for {
-			res, resp, err = g.client.GetPullRequest(g.context, g.options.GithubOrg, g.options.GithubRepo, pr)
-			if err != nil {
-				if !canWaitAndRetry(resp, err) {
-					return nil, err
-				}
-			} else {
-				break
-			}
+	// Given the PR number that we've now converted to an integer, get the PR from
+	// the API
+	mainPr, err := g.getPr(mainPrNum)
+	if err != nil {
+		return prs, err
+	}
+
+	prs = append(prs, mainPr)
+
+	originPrNum, origPrErr := originPrNumFromPr(mainPr)
+	if origPrErr == nil {
+		originPr, err := g.getPr(originPrNum)
+		if err == nil {
+			prs = append(prs, originPr)
 		}
-		prs = append(prs, res)
 	}
 
 	return prs, err
 }
 
-func prsNumForCommitFromMessage(commitMessage string) (prs []int, err error) {
+func (g *Gatherer) getPr(prNum int) (*gogithub.PullRequest, error) {
+	for {
+		res, resp, err := g.client.GetPullRequest(g.context, g.options.GithubOrg, g.options.GithubRepo, prNum)
+		if err != nil {
+			if !canWaitAndRetry(resp, err) {
+				return nil, err
+			}
+		} else {
+			return res, err
+		}
+	}
+}
+
+var (
 	// Thankfully k8s-merge-robot commits the PR number consistently. If this ever
 	// stops being true, this definitely won't work anymore.
-	regex := regexp.MustCompile(`Merge pull request #(?P<number>\d+)`)
-	pr := prForRegex(regex, commitMessage)
-	if pr != 0 {
-		prs = append(prs, pr)
+	regexMergedPR = regexp.MustCompile(`Merge pull request #(?P<number>\d+)`)
+
+	// If the PR was squash merged, the regexp is different
+	regexSquashPR = regexp.MustCompile(`\(#(?P<number>\d+)\)`)
+
+	// The branch name created by "hack/cherry_pick_pull.sh"
+	regexHackBranch = regexp.MustCompile(`automated-cherry-pick-of-#(?P<number>\d+)`)
+
+	// The branch name created by k8s-infra-cherrypick-robot
+	regexProwBranch = regexp.MustCompile(`cherry-pick-(?P<number>\d+)-to`)
+)
+
+func prNumForCommitFromMessage(commitMessage string) (prs int, err error) {
+	if pr := prForRegex(regexMergedPR, commitMessage); pr != 0 {
+		return pr, nil
 	}
 
-	regex = regexp.MustCompile(`automated-cherry-pick-of-#(?P<number>\d+)`)
```

Comment:
why is this gone?

Reply:
moved in `originPrNumFromPr`

Reply:
I see no harm in keeping both. There might be some projects depending on this format in the commit message.

Reply:
I don't see how that could happen.

Reply:
It's just a matter of backwards compatibility.
But up to the owners of the repo to decide.

Reply:
ok